packages/xsnap/test/test-xsnap.js

```diff
@@ -400,6 +402,101 @@ test('heap exhaustion: orderly fail-stop', async t => {
     const vat = xsnap({ ...xsnapOptions, meteringLimit: 0, debug });
     t.teardown(() => vat.terminate());
     // eslint-disable-next-line no-await-in-loop
-    await t.throwsAsync(vat.evaluate(grow));
+    await t.throwsAsync(vat.evaluate(grow), { message: /exited with code 1$/ });
+  }
+});
+
+test('property name space exhaustion: orderly fail-stop', async t => {
+  const grow = qty => `
+  const objmap = {};
+  try {
+    for (let ix = 0; ix < ${qty}; ix += 1) {
+      const key = \`k\${ix}\`;
+      objmap[key] = 1;
+      if (!(key in objmap)) {
+        throw Error(key);
+      }
+    }
+  } catch (err) {
+    // name space exhaustion should not be catchable!
```

Comment:
Good, but if this managed to catch the first few times, and then stopped catching it, we'd not be able to distinguish that failure mode from the intended one. Would it be possible to have the `catch` clause raise an exception that would cause the worker to exit with something measurably different (something other than `exited with code 6`) ?

If that's too much hassle, just having it print a "wait a minute I shouldn't be printed" and manually eyeball it once should be sufficient.

Reply:
I don't understand your point. The `try ... catch` is outside the `for` loop, so it catches either 0 or 1 times. If it catches, the `vat.evaluate` won't reject, so it'll fail `t.throwsAsync`.

In any case, I put a for-ever loop in the catch block to exit with "too much computation" in 0b4501f0a .